akka-http-testkit/src/test/java/akka/http/javadsl/testkit/JUnitRouteTestTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.javadsl.testkit;
+
+import java.util.function.Function;
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.Test;
+import akka.http.javadsl.testkit.*;
+
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.model.headers.*;
+import akka.http.javadsl.server.Rejections;
+
+public class JUnitRouteTestTest extends JUnitRouteTest {
+
+  @Test
+  public void testTheMostSimpleAndDirectRouteTest() {
+    TestRoute route =
+      testRoute(
+        complete(HttpResponse.create())
+      );
+
+    route.run(HttpRequest.GET("/"))
+      .assertStatusCode(StatusCodes.OK);
+  }
+
+  @Test
+  public void testUsingADirectiveAndSomeChecks() {
+    RawHeader pinkHeader = RawHeader.create("Fancy", "pink");
+
+    TestRoute route =
+      testRoute(
+        respondWithHeader(pinkHeader, () ->
+          complete("abc")
+        )
+      );
+
+    route.run(HttpRequest.GET("/").addHeader(pinkHeader))
+      .assertStatusCode(StatusCodes.OK)
+      .assertContentType("text/plain; charset=UTF-8")
+      .assertEntity("abc")
+      .assertHeaderExists("Fancy", "pink");
+  }
+
+  @Test
+  public void testProperRejectionCollection() {
+    TestRoute route =
+      testRoute(
+        get(() ->
+          complete("naah")
+        ).orElse(
+          put(() ->
+            complete("naah")
+          )
+        )
+      );
+
+    route.run(HttpRequest.POST("/abc").withEntity("content"))
+      /* XXX: Expected?
+      .assertRejections(
+        Rejections.method(HttpMethods.GET),
+        Rejections.method(HttpMethods.PUT)
+      )
```

Comment:
I think both kinds of testing are reasonable. Testing unsealed makes sense for testing reusable route components, while testing sealed makes sense for testing complete routes from the outside.

Hmm, I think we should try to keep the styles consistent between Scala and Java or make it explicit at least what you want to do. So, how about deprecating `run` in favor of `runSealed` and introducing another `runUnsealed` to introduce the Scala-like variant?